WordPress/src/main/java/org/wordpress/android/models/AccountModel.java

```diff
@@ -0,0 +1,135 @@
+package org.wordpress.android.models;
+
+import android.text.TextUtils;
+
+import org.json.JSONObject;
+
+public class AccountModel {
+    // General state
+    private boolean mIsWordPressComUser;
+
+    // WordPress.com only - data fetched from the REST API endpoint
+    private String mUserName;
+    private long mUserId;
+    private String mDisplayName;
+    private String mProfileUrl;
+    private String mAvatarUrl;
+    private long mPrimaryBlogId;
+    private int mSiteCount;
+    private int mVisibleSiteCount;
+    private String mAccessToken;
+
+    public AccountModel() {
+        init();
+    }
+
+    public void init() {
+        mIsWordPressComUser = false;
+        mUserName = "";
+        mUserId = 0;
+        mDisplayName = "";
+        mProfileUrl = "";
+        mAvatarUrl = "";
+        mPrimaryBlogId = 0;
+        mSiteCount = 0;
+        mVisibleSiteCount = 0;
+        mAccessToken = "";
+    }
+
+    public void updateFromRestResponse(JSONObject json) {
+        mIsWordPressComUser = true;
+        mUserId = json.optLong("ID");
+        mUserName = json.optString("username");
+        mDisplayName = json.optString("display_name");
+        mProfileUrl = json.optString("profile_URL");
+        mAvatarUrl = json.optString("avatar_URL");
+        mPrimaryBlogId = json.optLong("primary_blog");
+        mSiteCount = json.optInt("site_count");
+        mVisibleSiteCount = json.optInt("visible_site_count");
+    }
+
+    public long getUserId() {
+        return mUserId;
+    }
+
+    public void setUserId(long userId) {
+        mUserId = userId;
+    }
+
+    public void setPrimaryBlogId(long primaryBlogId) {
+        mPrimaryBlogId = primaryBlogId;
+    }
+
+    public long getPrimaryBlogId() {
+        return mPrimaryBlogId;
+    }
+
+    public String getUserName() {
+        return mUserName;
```

Comment:
I've found it helpful to have string getters use `StringUtils.notNullString()` [like this](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/2425-account-model/WordPress/src/main/java/org/wordpress/android/models/ReaderBlog.java#L87) to ensure that null is never returned (which cuts down on hidden NPEs). Is that worthwhile here as well?